### PR TITLE
chore: migrate angular e2e app to ng cli

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fix https://github.com/elastic/apm-agent-rum-js/issues/450
+ Follow up of #962 
+ Moves the E2E app to the Angular build system using `ng` CLI and packager. 
+ The build for E2E app is moved under `test/e2e/build` and then tests are ran to make sure our existing e2e test setup works and also `ng serve` works for local development. 

If you want to run the E2E app in local, you can try out using `ng serve app` which would run the build command and watch your files for local changes.

### CODE REVIEW

File: packages/rum-angular/tsconfig.lib.prod.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "extends": "./tsconfig.lib.json",
   "angularCompilerOptions": {
-    "enableIvy": true
+    "enableIvy": false
```
Comment (vigneshshanmugam) [inline packages/rum-angular/tsconfig.lib.prod.json:4]: Publishing fails when ivy is set to true. The recommendation from Angular team is to publish the libraries by keeping enableIvy as `false` and Applications using the library can opt in to ivy mode by using Angular Compatibility Compiler (ngcc). More details here https://next.angular.io/guide/creating-libraries#publishing-your-library

File: packages/rum-angular/test/e2e/with-router/app.e2e-spec.js [hunk 2]
```diff
@@ -26,29 +26,50 @@
...
+  })
 
-  it('should run angular app and capture route-change', function() {
+  it('should run angular app and capture route-change', async () => {
     /**
      * Should render home page on load
      */
-    const notFoundElement = $('app-root app-home h2')
-    expect(notFoundElement.getText()).toEqual('Home page')
+    const result = await browser.findElement(
```
Comment (vigneshshanmugam) [inline packages/rum-angular/test/e2e/with-router/app.e2e-spec.js:43]: the code on this file is converted to make sure it works for devtools automation protocol as well.

File: packages/rum-angular/package.json [hunk 3]
```diff
@@ -18,8 +18,12 @@
   },
   "scripts": {
     "build": "ng build --prod",
+    "build:e2e": "ng build app --prod",
     "test:unit": "ng test",
-    "test": "MODE=none npm run test:unit",
+    "test:e2e": "npm run script runE2eTests packages/rum-angular/wdio.conf.js",
+    "test:sauce": "npm run script runSauceTests packages/rum-angular true test:unit build:e2e test:e2e",
+    "test": "npm run test:sauce",
```
Comment (jahtalab) [inline packages/rum-angular/package.json:25]: Shouldn't this be running both `test:sauce` and `test:e2e`?

File: packages/rum-angular/angular.json [hunk 4]
```diff
@@ -26,8 +26,60 @@
...
+            "main": "test/e2e/with-router/main.ts",
+            "polyfills": "test/e2e/with-router/polyfills.ts",
+            "tsConfig": "test/e2e/with-router/tsconfig.app.json",
+            "aot": true,
+            "assets": [
+              "test/e2e/with-router/assets"
+            ]
+          },
+          "configurations": {
+            "production": {
```
Comment (jahtalab) [inline packages/rum-angular/angular.json:55]: Where this production configuration is used?

**Review by jahtalab — COMMENTED** [review]
Thanks @vigneshshanmugam ,

Generally looks good, however, there are a [few warnings](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-rum%2Fapm-agent-rum-mbp/detail/PR-974/12/pipeline/520#step-1298-log-64) in the main build script and also [in e2e build](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-rum%2Fapm-agent-rum-mbp/detail/PR-974/12/pipeline/1775#step-2216-log-95)

I'm also going to run it locally.

**Comment by vigneshshanmugam** [comment]
@jahtalab I have added fix for some of the warnings. 

> WARNING: 'NavigationStart' is imported from external module '@angular/router' but never used

For this to get fixed, we have to change to `import type { NavigationStart } from '@angular/router'` -> need to update typescript -> need to update prettier to `2.0` to understand the syntax which breaks the formatting locally. I will do a follow up PR to change the Eslint and prettier configs to get these sorted out.

**Comment by vigneshshanmugam** [comment]
@jahtalab I am merging this to our `feat-angular` branch. Lets fix the `NavigationStart` warning in the other PR #978

**Comment by jahtalab** [comment]
@vigneshshanmugam , there's also a warning about naming (I don't have the exact message since the logs are no longer accessible)

**Comment by vigneshshanmugam** [comment]
@jahtalab Fixing it with the next PR - https://github.com/vigneshshanmugam/apm-agent-js-base/blob/4be1c2f5240c5740f01e2794b86105dbbbe9f065/packages/rum-angular/ng-package.json. Waiting for the prettier to get merged.

**Pull request merged**